tests/chart_tests/test_nats_jetstream.py

```diff
@@ -68,6 +68,9 @@ def test_nats_statefulset_with_jetstream_and_tls(self, kube_version):
                 "keyFile": f"{jetStreamCertPrefix}-client/tls.key",
             },
         }
+        assert docs[7]["spec"]["template"]["spec"]["nodeSelector"] == {}
```

Comment:
We need a test case here to verify that custom nodeSelector, affinity, and tolerations work as expected.